lib/install/management/appliance.go

```diff
@@ -181,19 +181,39 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 		}
 	}
 
-	// Only retry VM destroy on CurrentAccess error
+	retryRemovalError := func(err error) bool {
+		return tasks.IsConcurrentAccessError(err) || tasks.IsInvalidPowerStateError(err)
+	}
+
+	// Only retry VM destroy on ConcurrentAccess error
+	var destroyFirstAttempt bool
 	err = retry.Do(func() error {
+		if !destroyFirstAttempt {
+			d.op.Debugf("detected an OOB powerOn for container (%s) attempting an additional shutdown before removal", vm.Reference().String())
+			vm.PowerOff(d.op)
```

Comment:
In light of this code. Another path is to stop trying to power everything down right away and move to a power off case by case. Thoughts?

Reply:
idk how possible this case is, but what if at first attempt, we power off the VM, destroy fails with `ConcurrentAccess`, we retry again after several seconds, an OOB power-on happens during the time. Since it's not first attempt, we go ahead and delete, it fails with `InvalidPowerState` again?
what if we do something like an additional check in the retry function, if it's invalid power state, we power off, otherwise we do nothing

Reply:
idk how possible this is, but what if at first attempt, we power off the vm, and later delete fails with `ConcurrentAccess`. Then we retry after several seconds, during which an OOB power on occurred, and second delete fails with `InvalidPowerState`
do we want to do something like a conditional check in the retry function, for ex if the error is `ConcurrentAccess`, we power off the VM, otherwise do nothing

Reply:
That case is plausible, but in our case what will happen now is that we will retry on the concurrent access and then we will retry on the invalid power state and during the retry decision process attempt to power off the vm again. So that case should be covered now :)